consensus/vbft/block_pool.go

```diff
@@ -147,6 +147,15 @@ func (pool *BlockPool) newBlockProposal(msg *blockProposalMsg) error {
 
 	// add msg to proposals
 	candidate.Proposals = append(candidate.Proposals, msg)
+
+	// add endorse-sig
+	proposer := msg.Block.getProposer()
+	eSig := &CandidateEndorseSigInfo{
+		EndorsedProposer: proposer,
+		Signature:        msg.Block.Block.Header.SigData[0],
```

Comment:
where is the code that guarantee the proposer's signature is the first one  when msg comes from faulty node?

Reply:
2f+1 voting, should include proposer.
proposal msg dont need to be the first one.

Reply:
https://github.com/ontio/ontology/blob/master/consensus/vbft/msg_types.go#L72